packages/wp-source/src/actions.ts

```diff
@@ -65,8 +65,22 @@ const actions: WpSource["actions"]["source"] = {
     try {
       let { route } = linkParams;
       // Transform route if there is some redirection.
-      const redirection = getMatch({ route, link }, libraries.source.redirections);
-      if (redirection) route = redirection.func(redirection.params);
+      const redirection = getMatch(
+        { route, link },
+        libraries.source.redirections
+      );
+
+      // Create a variable that holds the value of the link that we will later
+      // pass to the `getMatch()` function. If there is a redirection, we update
+      // the value of `route`, so we'll have to later derive the link again
+      // based on the "new" (redirected) route.
+      let linkForHandler = link;
+      if (redirection) {
+        route = redirection.func(redirection.params);
+
+        // Derive the link from the "redirected" route.
+        linkForHandler = normalize(route);
+      }
```

Comment:
Is there a case were we could capture this within a test? 🤔

Reply:
Yeah, I guess so. In the `3xx-redirections.tests.ts` file, there is a `handler.func` mock that we can use to check what params were passed, and if `link` has the correct value.

@michalczaplinski, could you check the `link` value in those tests?

Reply:
yup, I've added a test case. Thanks for pointing it out @DAreRodz and @cristianbote 🙂 

I explain in a new video why I did it this way: https://www.loom.com/share/4e265877436547848d6df9cd32cddd18

Reply:
Oops, my bad, hahaha :see_no_evil: You're completely right. I don't know why I thought about 3xx redirections. :sweat_smile:

Reply:
Wait a second.

If this is about internal redirections, then the value of `link` should not be changed! :worried: Or should it be? I'm confused now. :dizzy_face:

Reply:
I think Michal is right, the redirections only change `route` so they are not useful for handlers with RegExps.

For example, imagine there is

- A redirection of the posts archive from `/blog/` to `/`.
- A handler that uses this `RegExp:\\/?home_option=.*` which should match the posts archive with the `home_option` query.

The handler won't capture the link `/blog/?home_option=xxx` because only the `route` is changed and the RegExp is using `link` inside `getMatch()`.

What I don't know is if the proposed solution is enough to fix the problem. If I am not mistaken, some of the redirections are removing the queries.

```js
if (postsPage) {
  const pattern = concatLink(subdirectory, postsPage);
  redirections.push({
    name: "posts page",
    priority: 10,
    pattern,
    func: () => "/",
  });
}
```

Those were not important in the `path-to-regexp` patterns, but they are now in the RegExps. So maybe to solve this we also need to take a look at the redirections.

What do you think guys?

Reply:
Aaah, gotcha! Sorry again, I missunderstood the code. :see_no_evil: 

I thought the `link` property was passed later to the handler function, but it's only passed to the `getMatch()` call.